resources/posts/2021/2021-11-15-introducing-light-kernel-for-symfony-console-apps.md

```diff
@@ -0,0 +1,200 @@
+use Symplify\SymplifyKernel\ContainerBuilderFactory;
+use Symplify\ComposerJsonManipulator\ValueObject\ComposerJsonManipulatorConfig;
+
+final class MonorepoBuilderKernel
+{
+    /**
+     * @param string[] $configFiles
+     */
+    public function createFromConfigs(array $configFiles): ContainerInterface
+    {
+        // provide local config here
+        $configFiles[] = __DIR__ . '/../../config/config.php';
+
+        // external configs
+        $configFiles[] = ComposerJsonManipulatorConfig::FILE_PATH;
+
+        $containerBuilderFactory = new ContainerBuilderFactory();
+
+        $containerBuilder = $containerBuilderFactory->create($configFiles, [], []);
+
+        // build the container
+        $containerBuilder->compile();
+
+        return $containerBuilder;
+    }
+}
+```
+
+<br>
+
+How would it meet our requirements?
+
+* single method <span class="text-success pt-3 pb-3">✅</span>
+* only `symfony/dependency-injection` <span class="text-success pt-3 pb-3">✅</span>
+* no http <span class="text-success pt-3 pb-3">✅</span>
+
+## How to use ContainerBuilderFactory in 3 Steps
+
+For a couple of years, the [Symplify](https://github.com/symplify/symplify) uses own Symfony Kernel wrapper package - the `symplify/symplify-kernel`. It abstracts repeated methods and eases testing. What better place we could use for adding a dependency container factory?
+
+<br>
+
+**1. Install Symplify Kernel**
+
+```bash
+composer require symplify/symplify-kernel
+```
+
+<br>
+
+**2. Extend `AbstractSymplifyKernel` and provide config files**
+
+This the full kernel for the [EasyCI package](/blog/5-commands-from-easy-ci-that-makes-your-ci-stronger/) looks like:
+
+```php
+namespace Symplify\EasyCI\Kernel;
+
+use Psr\Container\ContainerInterface;
+use Symplify\Astral\ValueObject\AstralConfig;
+use Symplify\ComposerJsonManipulator\ValueObject\ComposerJsonManipulatorConfig;
+use Symplify\SymplifyKernel\HttpKernel\AbstractSymplifyKernel;
+
+final class EasyCIKernel extends AbstractSymplifyKernel
+{
+    /**
+     * @param string[] $configFiles
+     */
+    public function createFromConfigs(array $configFiles): ContainerInterface
+    {
+        $configFiles[] = __DIR__ . '/../../config/config.php';
+        $configFiles[] = ComposerJsonManipulatorConfig::FILE_PATH;
+        $configFiles[] = AstralConfig::FILE_PATH;
+
+        return $this->create([], [], $configFiles);
+    }
+}
+```
+
+<br>
+
+**3. Boot Kernel in your bin file and enjoy the Symfony DI**
+
+```php
+$easyCIKernel = new EasyCIKernel();
+$easyCIKernel->createFromConfigs([__DIR__ . '/config/config.php']);
+
+$container = $easyCIKernel->getContainer();
+
+/** @var Application $application */
+$application = $container->get(Application::class);
+exit($application->run());
+```
+
+That's it! In the end, the setup is very simple.
+
+It allowed us to drop the following files from 4 Simplify CLI packages:
+
+<img src="/assets/images/posts/2021/light_kernel_vendor_diff.gif" class="img-thumbnail">
```

Comment:
this path seems not exists now?

Reply:
I see, it still there https://github.com/TomasVotruba/tomasvotruba.com/blob/main/public/assets/images/posts/2021/light_kernel_vendor_diff.gif